master/internal/experiment_filter.go

```diff
@@ -404,30 +406,26 @@ func (e experimentFilter) toSQL(q *bun.SelectQuery,
 			} else {
 				q.Where(queryString, queryArgs...)
 			}
-		case projectv1.LocationType_LOCATION_TYPE_TRAINING.String():
+		case projectv1.LocationType_LOCATION_TYPE_TRAINING.String(),
+			projectv1.LocationType_LOCATION_TYPE_CUSTOM_METRIC.String():
 			queryColumnType := projectv1.ColumnType_COLUMN_TYPE_UNSPECIFIED.String()
 			if e.Type != nil {
 				queryColumnType = *e.Type
 			}
-			metricDetails := strings.Split(e.ColumnName, ".")
-			metricQualifier := metricDetails[len(metricDetails)-1]
-			metricName := strings.TrimSuffix(
-				strings.TrimPrefix(e.ColumnName, "training."),
-				"."+metricQualifier)
-			if !slices.Contains(SummaryMetricStatistics, metricQualifier) {
-				return nil, status.Errorf(codes.InvalidArgument,
-					"sort training metrics by statistic: last, max, min, or mean")
+			metricGroup, metricName, metricQualifier, err := parseMetricsName(e.ColumnName)
+			if err != nil {
+				return nil, err
 			}
 			var col string
 			var queryArgs []interface{}
 			var queryString string
 			if metricQualifier == "mean" {
-				locator := bun.Safe("trials.summary_metrics->'avg_metrics'")
+				locator := bun.Safe(fmt.Sprintf("trials.summary_metrics->'%s'", metricGroup))
 				col = fmt.Sprintf(`(?->?->>'sum')::float8 / (?->?->>'count')::int`)
```

Comment:
this mean is susceptible to overflow. probably need to use a built-in to get it right.

Reply:
imo, the precomputed summary metrics should offer mean.